hathor/cli/run_node.py

```diff
@@ -49,6 +49,8 @@ def create_parser(self) -> ArgumentParser:
         parser.add_argument('--recursion-limit', type=int, help='Set python recursion limit')
         parser.add_argument('--allow-mining-without-peers', action='store_true', help='Allow mining without peers')
         parser.add_argument('--min-block-weight', type=int, help='Minimum weight for blocks')
+        parser.add_argument('--fast-init', action='store_true',
+                            help='Skip some transaction validations when initializing the node')
```

Comment:
I don't see the benefit of this not being the default. I'd rather have a `--full-verification` flag, with reversed meaning. Specially if the speedup from this is really significant. Particularly, if the concern is to protect ourselves from maliciously altered storage data, that wouldn't matter, because then they could as well alter the node code so it's skipped anyway. Setups where this would matter (external storage) could take this into account and add the `--full-verification` flag.

To add to this, defaulting to faster startup seem to be the norm (like Bitcoin only verifying the last 6 blocks by feault).

Since @msbrogli suggested this, what do you think?

Reply:
@jansegre But it is not so easy to temper with blocks/transactions because you'd have to solve the PoW of the whole DAG after it. On the other side, it is very easy to temper with the metadata, and it would affect the consensus.

Anyway, I agree that I myself would run with `--fast-init` most of the times. So, I agree that we can have it by default. I'm only concerned people would never run a full verification, and it can be dangerous.

Maybe we should add a counter of "number of initializations without full verification" and warn users from time to time to run a full verification, just like linux partitions do.

Reply:
> @jansegre But it is not so easy to temper with blocks/transactions because you'd have to solve the PoW of the whole DAG after it. On the other side, it is very easy to temper with the metadata, and it would affect the consensus.

My point was, that even easier than tempering with the metadata, is tempering with the code. Assuming one has disk access.

> Maybe we should add a counter of "number of initializations without full verification" and warn users from time to time to run a full verification, just like linux partitions do.

That sounds good too. We should discuss this in a separate issue.

Reply:
@msbrogli do you agree to create another issue to discuss this counter of "number of initializations without full verification"?

Reply:
Created the issue as we've discussed. https://github.com/HathorNetwork/hathor-core/issues/122